### DISCUSSION TITLE
UCISD 4-fold degeneracy problem.

### DISCUSSION THREAD

**Discussion by sunchong137** [pyscf/pyscf#2678] [Q&A] [answered]
Hi there, 

I am using the ` cisdvec_to_amplitudes` function in ucisd.py. I found that when transforming the `myci.ci` to the amplitudes, the 4-fold degeneracy is not considered for the double excitations within the same spins. Therefore, if we directly take the value returned: `c0, c1, c2 = myci.cisdvec_to_amplitudes(myci.ci)`  and construct the CISD wavefunction, the wavefunction is incorrect.  Therefore, I need to manually divide `c2[0]` and `c2[2]` by 4.

I wonder if this is the convention or something to be fixed.

Thank you for your time!

**Comment by sunqm** [comment, marked as answer]
The two coefficients have utilized the convention `i>=j, a>=b`, as shown in https://github.com/pyscf/pyscf/blob/master/pyscf/ci/ucisd.py#L351  . After applying the contraction `H*c`, the coefficients elements of `i<j or a<b` are removed by the `amplitudes_to_cisdvec` function.

**Comment by verena-neufeld** [comment]
Hi @sunchong137 , thanks for raising this issue. Has it been resolved now?
